subprojects/core/src/main/java/org/gradle/api/tasks/JavaExec.java

```diff
@@ -337,9 +372,16 @@ public FileCollection getClasspath() {
      */
     public JavaExec copyTo(JavaForkOptions options) {
         javaExecHandleBuilder.copyTo(options);
+        copyJvmArgProvidersTo(options);
```

Comment:
Shouldn't we copy `argProviders`, too?

Reply:
What do you mean? There is no place for argument providers on `JavaForkOptions`.
I chose to not support a `copyTo` to a `ConfigurableJavaForkOptions`, since this should not be necessary.